src/main/java/team/unnamed/dependency/resolve/SubDependenciesResolver.java

```diff
@@ -13,7 +25,47 @@
 
     @Override
     public List<Dependency> resolve(Dependency dependency) {
-        return null;
-    }
 
+        Set<String> repositories = new HashSet<>();
+
+        repositories.add(MavenDependency.CENTRAL);
+        ErrorDetails errorDetails = new ErrorDetails("Cannot resolve dependencies");
+        Set<Dependency> dependencies = new HashSet<>();
+        String pomUrl = dependency.getPossibleOriginUrls()[0];
+        try {
+            URL url = new URL(pomUrl);
+            HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
+            connection.setRequestProperty("User-Agent", "ddownloader");
+            try (BufferedInputStream inputStream = new BufferedInputStream(connection.getInputStream())) {
+                Model model = new MavenXpp3Reader().read(inputStream);
+                for (Repository repository : model.getRepositories()) {
+                    repositories.add(repository.getUrl());
+                }
+                for (org.apache.maven.model.Dependency mavenDependency : model.getDependencies()) {
+                    if (!mavenDependency.getScope().equalsIgnoreCase("compile")) continue;
+                    String version = mavenDependency.getVersion();
+                    Dependency subDependency = new MavenDependency(
+                            repositories.toArray(new String[]{}),
+                            mavenDependency.getGroupId(),
+                            mavenDependency.getArtifactId(),
+                            version,
+                            mavenDependency.getArtifactId().toLowerCase() + ".jar",
+                            false
+                    );
+                    dependencies.add(subDependency);
+
+                }
+            } catch (XmlPullParserException e) {
+                errorDetails.add(e);
+            }
+        } catch (IOException e) {
+            errorDetails.add(e);
+        }
+
+        if (errorDetails.errorCount() > 0) {
+            throw errorDetails.toResolveException();
+        }
+
+        return new ArrayList<>(dependencies);
+    }
```

Comment:
Including and using maven-model dependency isn't necessary, simply use the JDK XML parser